apps/src/music/player/MusicPlayer.js

```diff
@@ -215,6 +217,20 @@ export default class MusicPlayer {
     );
   }
 
+  // Returns the current measure, with a fractional component representing the
+  // exact playhead.
+  // Returns -1 if music is not playing.
+  getCurrentMeasureExact() {
```

Comment:
Can we just use `getPlayheadPosition` here?

Reply:
Thanks for poking on this.  I'd been going back and forth on it, and did some more, and came up with a slightly new approach.  Playhead isn't enough information by itself, because we need to know how we scale that to pixels, and `Timeline` is currently the component responsible for that.  In the end, this new `getCurrentMeasureExact` (where `Exact` means that it can have a fractional part, and could probably have a clearer name) does the job.  `Timeline` grabs it at the begining of its render, and then the element uses it to determine whether it's currently playing.  (I did consider having the element just compare the element's rendered location and width to the rendered location of the playhead, but it feels more robust to compare the element's `when` and `length` to the current measure.)

Separately, it might be a longer term improvement to extract the code that scales timeline to pixels out of `Timeline`.

Reply:
Good question.  Here's one attempt at a new approach.

It seems handy to have a `getCurrentMeasure` for interpreted code to get a 1-based current measure.  Separately, the timeline renderer wants a 0-based current measure, so I added `getCurrentMeasureInternal` which leverages the former's implementation.

The Timeline grabs the 0-based current measure once at the beginning of its render, and the timeline elements use that value to determine whether they are currently playing.

Reply:
Ahh, right, I forgot that one uses a 1-based timestamp vs a 0-based. I do think what you have is a better direction (though I think for me, `getPlayheadPosition` is still more appropriate for an "exact" timing, vs I'd expect `getCurrentMeasure` to return the current whole number measure). 

That being said, I think this points to us generally needing to tighten up our contracts around measure and playhead timing a bit (probably out of scope of this PR, but something we should do sooner rather than later). I can see this potentially causing some unwanted issues down the line in case the underlying assumptions are forgotten. Thinking about it a bit, I think my inclination would be to treat everything outside of `MusicPlayer` as 1-based, with functions like `getPlayheadTime`, `getCurrentMeasure`, etc, and have all UI elements adjust accordingly. The conversion from 1-based to 0-based would then just be contained to `MusicPlayer` which would only happen when playing or queueing elements (we could even break off the sample playing/queueing logic into a separate component, especially as we start to think about the distinctions between for example drum sequences and the individual samples they contain). What do you think?

Reply:
Made some more updates after our offline discussion.  Notably, the `MusicPlayer` contract is more consistent: its  `getCurrentPlayhead` becomes the normal way to get the 1-based, measure-scaled current playback position, and its `soundEvents` also now use 1-based, measure-scaled `when` times.  Separately, the `Timeline` is now given the value of `getCurrentPlayhead` each time it is rendered.